File: src/Grid/App/Component/Assets/AssetModel.js
function Asset(data) {
    var self = ModelBase(this);
    self.modelState = ko.validatedObservable();
    self.isValid = function () {
        return self.modelState.isValid();
    };
    self.showErrors = function () {
        self.modelState.errors.showAllMessages(true);
    };


    self.id = ko.observable(data.Id || 0);
    self.name = ko.observable(data.Name);
    self.title = ko.observable(data.Title).extend({ required: { params: true, message: messages.general.Title } });
    self.description = ko.observable(data.Description);
    self.serialNumber = ko.observable(data.SerialNumber);
    self.tagNumber = ko.observable(data.TagNumber);
    self.specifications = ko.observable(data.Specifications);
    self.brand = ko.observable(data.Brand);
    self.cost = ko.observable(data.Cost);
    self.modelNumber = ko.observable(data.ModelNumber);
    self.isBrandNew = ko.observable(data.IsBrandNew);
    self.purchaseDate = ko.observable(self.parseDate(data.PurchaseDate) || "");
    self.warrantyExpiryDate = ko.observable(self.parseDate(data.WarrantyExpiryDate) || "");
    self.assetCategoryId = ko.observable(data.AssetCategoryId).extend({ required: { params: true, message: messages.general.Select } });
    self.assetCategory = ko.observable();
    self.departmentId = ko.observable(data.DepartmentId).extend({ required: { params: true, message: messages.general.Select } });
    self.department = ko.observable();
    self.vendorId = ko.observable(data.VendorId).extend({ required: { params: true, message: messages.general.Select } });
    self.vendor = ko.observable();
    self.allocatedEmployeeId = ko.observable(data.AllocatedEmployeeId);
    self.allocatedEmployee = ko.observable();
    self.state = ko.observable(data.State).extend({ required: { params: true, message: messages.asset.State } });
    self.stateType = ko.observable();

    if (data.State == 0) {
        self.stateType("Active");
    }
    else if (data.State == 1) {
        self.stateType("InStore");
    } else if (data.State == 2) {
        self.stateType("UnderRepair");
    }
    else if (data.State == 3) {
        self.stateType("Dead");
    } else if (data.State == 4) {
        self.stateType("Unknown");
    }

    self.state.subscribe(function (newValue) {
        if (newValue == '1' || newValue == '2' || newValue == '3' || newValue == '4') {
            self.allocatedEmployeeId('');
            self.allocatedEmployee(null)
        }       
    });


    if (data.AssetCategory) {
        self.assetCategory(data.AssetCategory.Title);
    }
    if (data.Department) {
        self.department(data.Department.Title);
    }
    if (data.Vendor) {
        self.vendor(data.Vendor.Title);
    }
    if (data.AllocatedEmployee) {
        if (data.AllocatedEmployee.User) {
            if (data.AllocatedEmployee.User.Person) {
                if (data.State != 3)
                self.allocatedEmployee(data.AllocatedEmployee.User.Person.Name);
            }
        }
    }


    self.modelState = ko.validatedObservable(
{
    Title: self.title,
    AssetCategoryId: self.assetCategoryId,
    State: self.state,
    DepartmentId: self.departmentId,
    VendorId: self.vendorId,
});
    self.resetValidation = function () {
        self.title.isModified(false);
        self.assetCategoryId.isModified(false);
        self.state.isModified(false);
        self.departmentId.isModified(false);
        self.vendorId.isModified(false);
    };
};

File: src/Grid.Api/Models/IMS/AssetModel.cs
using Grid.Features.IMS.Entities;
using Grid.Features.IMS.Entities.Enums;
using System;

namespace Grid.Api.Models.IMS
{
    public class AssetModel : ApiModelBase
    {
        public string SerialNumber { get; set; }
        public string TagNumber { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string Specifications { get; set; }
        public string BrandAndModel { get; set; }
        public string Brand { get; set; }
        public decimal Cost { get; set; }
        public string ModelNumber { get; set; }
        public bool IsBrandNew { get; set; }
        public string StateType { get; set; }
        public AssetState State { get; set; }
        public DateTime? PurchaseDate { get; set; }
        public DateTime? WarrantyExpiryDate { get; set; }
        public int AssetCategoryId { get; set; }
        public string AssetCategory { get; set; }
        public int? DepartmentId { get; set; }
        public string Department { get; set; }
        public int? VendorId { get; set; }
        public string Vendor { get; set; }
        public string TitleAndModel { get; set; }
        public int? AllocatedEmployeeId { get; set; }
        public string AllocatedEmployee { get; set; }
        public int? AllocatedByEmployeeId { get; set; }
        public string AllocatedByEmployee { get; set; }
        public DateTime? AllocatedOn { get; set; }

        public AssetModel()
        {

        }
        public AssetModel(Asset asset)
        {
            Id = asset.Id;

            if (asset.State != AssetState.Dead)
            {
                if (asset.AllocatedEmployee?.User.Person != null)
                {
                    AllocatedEmployee = asset.AllocatedEmployee.User.Person.Name;
                }
            }
            if (asset.Department != null)
            {
                Department = asset.Department.Title;
            }
            if (asset.AssetCategory != null)
            {
                AssetCategory = asset.AssetCategory.Title;
            }
            if (asset.Vendor != null)
            {
                Vendor = asset.Vendor.Title;
            }

            TitleAndModel = $"{asset.Title} {asset.ModelNumber}";
            TagNumber = asset.TagNumber;
            SerialNumber = asset.SerialNumber;
            StateType = GetEnumDescription(asset.State);
            Specifications = asset.Specifications;
            BrandAndModel = $"{asset.Brand} {asset.ModelNumber}";
            Title = asset.Title;
            Cost = asset.Cost;
            Description = asset.Description;
            Brand = asset.Brand;
            ModelNumber = asset.ModelNumber;
            IsBrandNew = asset.IsBrandNew;
            PurchaseDate = asset.PurchaseDate;
            WarrantyExpiryDate = asset.WarrantyExpiryDate;
        }
        public AssetModel(AssetAllocation asset)
        {
            Id = asset.Id;

            if (asset.AllocatedEmployee?.User.Person != null)
            {
                AllocatedEmployee = asset.AllocatedEmployee.User.Person.Name;
            }
            StateType = GetEnumDescription(asset.State);
            AllocatedOn = asset.AllocatedOn;
        }
    }
}
